server/src-lib/Hasura/GraphQL/Transport/HTTP/Protocol.hs

```diff
@@ -99,48 +98,18 @@ encodeGQErr :: Bool -> QErr -> J.Value
 encodeGQErr includeInternal qErr =
   J.object [ "errors" J..= [encodeGQLErr includeInternal qErr]]
 
-data GQResult a
-  = GQSuccess !a
-  | GQPreExecError ![J.Value]
-  | GQExecError ![J.Value]
-  deriving (Show, Eq, Functor, Foldable, Traversable)
+type GQResult a = ExceptT GQExecError Identity a
```

Comment:
`ExceptT` over `Identity` is basically just an awkward version of `Either`. Is there a reason we need `ExceptT` here specifically?

Reply:
Done!